lib/fluent/plugin/out_sumologic.rb

```diff
@@ -195,36 +195,20 @@ def sumo_key(sumo_metadata, chunk)
     source_host = sumo_metadata['host'] || @source_host
     source_host = extract_placeholders(source_host, chunk) unless source_host.nil?
 
-    "#{source_name}:#{source_category}:#{source_host}"
+    fields = sumo_metadata['fields'] || ""
+    fields = extract_placeholders(fields, chunk) unless fields.nil?
+
+    "#{source_name}:#{source_category}:#{source_host}:#{fields}"
```

Comment:
not addressed by this PR, but something we should fix: having a colon (`:`) in any metadata field or value would currently break this

Reply:
good point - I believe you can use ruby hashes as keys? maybe that's something we can look into though it does make it pretty complex